forge/test/mlir/mnist/training/test_training.py

```diff
@@ -91,3 +93,39 @@ def test_mnist_training():
             break
 
     print(f"Test (total) loss: {test_loss}")
+
+@pytest.mark.parametrize("batch_size", [1, 2, 16])
+@pytest.mark.parametrize("in_features", [784])
+@pytest.mark.parametrize("out_features", [10])
+def test_batch_size(batch_size, in_features, out_features):
+    class SimpleModel(nn.Module):
+        def __init__(self):
+            super(SimpleModel, self).__init__()
+            self.linear = nn.Linear(in_features, out_features)
+
+        def forward(self, x):
+            y = self.linear(x)
+            return nn.functional.softmax(y, dim=-1)
+    
+    in_data = torch.rand(batch_size, in_features)
+    out_data = torch.randint(0, out_features, (batch_size,))
+    target = nn.functional.one_hot(out_data, num_classes=out_features).float()
+
+    model = SimpleModel()
+
+    loss_fn = torch.nn.CrossEntropyLoss()
+    optimizer = torch.optim.SGD(model.parameters(), lr=0.001)
+    tt_model = forge.compile(model, sample_inputs=[torch.rand(batch_size, in_features)], loss=loss_fn, optimizer=optimizer)
+
+    optimizer.zero_grad()
+    
+    pred = tt_model(in_data)[0]
+    golden_pred = model(in_data)
+    compare_with_golden(golden_pred, pred)
```

Comment:
This will silently error out. You need to use `assert` here :))

Same comment for other samples

Reply:
> @nvukobratTT I think it would make more sense if we assert inside of this function...

I'm not sure if that is a better path at the moment..

More precisely, we need to define verification mechanics similar to the one we had on Buda (verification pipeline) where everything is encapsulated (data generation, asserts, etc.). 

In this case, this function is just one of the steps that we'll require there. Thus, having more flexibility around what we do with the output seems more meaningful at this point. 

Do you have some strong opinions on why we should do otherwise?

Reply:
No strong opinions, I just prefer this approach, since it reduces chances for silent failures.

It seems to me that we can do both, first encapsulate the assert directly into the `compare_with_golden()` until the verification pipeline is in-place, then do whatever is needed...